cert/cert.go

```diff
@@ -120,25 +130,51 @@ func (ca *CA) Refresh() (bool, error) {
 		return false, err
 	}
 
-	if bytes.Equal(cert, ca.pem) {
+	isCACertSame, err := CompareCertificates(cert, ca.pem)
+	if err != nil {
+		log.Infof("cert: error comparing CA certificates")
+		return false, err
+	}
+
+	if isCACertSame {
 		if ca.File != nil {
 			log.Infof("cert: existing CA certificate at %s is current", ca.File.Path)
 		}
 		return false, nil
 	}
 
+	// If CA cert has changed, write out new CA cert
 	if ca.File != nil {
 		err = ca.writeCert(cert)
 	}
 	// If there were no errors, update our internal notion of what the CA is.
 	if err != nil {
-		// see CA.Load(); strip prefix/trailing whitespace to ensure our bytes.Equal() checks don't false positive.
-		ca.pem = []byte(strings.TrimSpace(string(cert[:])))
+		ca.pem = cert
 	}
-
 	return true, err
 }
 
+// CompareCertificates x509 compares two CA certificates
+func CompareCertificates(cert1, cert2 []byte) (bool, error) {
+	p1, _ := pem.Decode(cert1)
```

Comment:
Any reason we don't pass the error returned to errors.New() below?

Same question for the other pem.Decode(cert2) invocation.

Reply:
Yep! pem.Decode doesn't actually return an error... it returns a block and []byte. If the block is nil (or the []byte is not nil) then the []byte was not properly parsed.